src/oc_erchef/apps/oc_chef_authz/src/oc_chef_authz_scoped_name.erl

```diff
@@ -0,0 +1,495 @@
+         find_group_authz_ids/2,
+
+         make_sql_callback/0,
+         initialize_context/1,
+         initialize_context/2,
+         make_name/2,
+         org_id_to_name/1
+        ]).
+
+-export([parse_scoped_name/3]).
+
+-ifdef(TEST).
+-compile([export_all]).
+-endif.
+
+%%
+%% Process names with scoping descriptor.
+%%
+
+%% This is derived from oc_chef_wm/src/oc_chef_wm_groups.erl; investigate what it will take to refactor this.
+-define(NAME, "[a-z0-9\-_]").
+
+-define(SCOPE_SEPARATOR, <<"::">>).
+-define(SCOPED_NAME_REGEX, "^(?:(" ?NAME "+)|(?:(" ?NAME "*)\\:\\:(" ?NAME "+)))$").
+
+-type db_callback() :: fun((any()) -> any()).
+
+-record(context, {org_id :: binary(),
+                  db_context :: any() | undefined,
+                  db_callback_fun :: db_callback()
+                 }).
+
+-record(sname, {full :: binary(),
+                base :: binary() | undefined,
+                org  :: binary() | 'global_org' | undefined,
+                org_id :: binary() | undefined,
+                authz_id :: binary() | undefined
+               }).
+
+-type lookup_type() :: 'actor' | 'user' | 'client' | 'group'.
+
+%%
+%% Context is used to thread external state information through the system
+%%
+%% Dialyzer is grouchy unless undefined is included in types.
+-spec initialize_context(binary()) -> #context{}.
+initialize_context(OrgId) ->
+    initialize_context(OrgId, make_sql_callback()).
+
+-spec initialize_context(binary(), db_callback()) -> #context{}.
+initialize_context(OrgId, CallBackFun) ->
+    initialize_context(OrgId, undefined, CallBackFun).
+
+-spec initialize_context(binary(), tuple() | undefined, db_callback()) -> #context{}.
+initialize_context(OrgId, DbContext, CallBackFun) ->
+    #context{org_id = OrgId,
+             db_context = DbContext,
+             db_callback_fun = CallBackFun}.
+
+%%
+%% Takes scoped names to authz ids
+%%   Names: list of names as binary; can include scoping symbol
+%%   Type: The type of object to look for
+%%   OrgContext: The name of the org whose context is local or 'global'
+%%
+%% We perform a series of successive lowerings;
+%% * parsing the name into base and org components
+%%   (if the later is present, otherwise inserting the current in-context org)
+%% * Mapping the org name to id
+%% * Looking up the org id, name pair in the appropriate table(s)
+
+%%
+%% Output:
+%% { [{Name, AuthzId}, [{Name, ErrorType}] }
+%%
+%%
+-spec names_to_authz_id(lookup_type() ,[binary()], #context{}) -> { [binary()],[{atom(),binary()}] }.
+names_to_authz_id(Type, Names, MapperContext) ->
+    %% Lower to fully qualified orgname, name
+    ScopedNames = parse_scoped_names(Names, is_scoped_type(Type), MapperContext),
+    {ProperNames, Errors} = lists:foldl(fun filter_errors/2, {[], []}, ScopedNames),
+
+    OrgCache = init_org_name_cache(),
+
+    %% Map org names to org ids
+    {NamesWithOrgIds, Errors2, _} =
+        lists:foldl(fun(Name, Acc) -> lookup_org_id(Name, Acc, MapperContext) end,
```

Comment:
I might be missing something but why do we pass the `MapperContext` to this lookup fun if it doesn't use it?